app/src/main-process/main.ts

```diff
@@ -121,11 +121,22 @@ process.on('uncaughtException', (error: Error) => {
   uncaughtException(error)
 })
 
+let willQuit = false
+
 if (__WIN32__ && process.argv.length > 1) {
-  if (handleSquirrelEvent(process.argv[1])) {
-    app.quit()
+  const arg = process.argv[1]
+  const promise = handleSquirrelEvent(arg)
+  if (promise) {
+    willQuit = true
+    promise
```

Comment:
We can't `await` in the topmost scope, sadly.

>it seems this would be more readable if it returned a promise yielding true or false depending on whether it was handled or not.

I agree, but we need to synchronously set the `willQuit` variable to avoid doing anything in the `ready` handler later on. This is the fix to https://github.com/desktop/desktop/issues/1875. I'm open to any ideas you have on that.